webpack.config.js

```diff
@@ -589,36 +499,34 @@ if (targetIsTest) {
 }
 
 if (targetIsDevServer) {
+    const proxyToServer = {
+        logLevel: 'silent',
+        target: process.env.MM_SERVICESETTINGS_SITEURL ?? 'http://localhost:8065',
+        xfwd: true,
+    };
+
     config = {
         ...config,
         devtool: 'eval-cheap-module-source-map',
         devServer: {
             liveReload: true,
-            proxy: [{
-                context: () => true,
-                bypass(req) {
-                    if (req.url.indexOf('/api') === 0 ||
-                        req.url.indexOf('/plugins') === 0 ||
-                        req.url.indexOf('/static/plugins/') === 0 ||
-                        req.url.indexOf('/sockjs-node/') !== -1) {
-                        return null; // send through proxy to the server
-                    }
-                    if (req.url.indexOf('/static/') === 0) {
-                        return path; // return the webpacked asset
-                    }
-
-                    // redirect (root, team routes, etc)
-                    return '/static/root.html';
+            proxy: {
```

Comment:
These changes to the proxy config were what fixed `make dev` itself. Something that we were doing before was making it so that the dev-server's websocket was repeatedly reconnecting and causing the page to refresh.

This is hopefully also clearer since I think we were misusing some of these options. Instead of having the proxy intercept all requests (by having `context` always return true) and then us telling Webpack where the requests should go (with `bypass`), the proxy is just handling requests that need to go to the MM server on the routes listed here and then the `historyApiFallback` option is being used to tell the app to redirect any URL that would 404 to the `root.html` instead.